go/vt/vtgate/planbuilder/testdata/filter_cases.json

```diff
@@ -4110,7 +4110,7 @@
           "Sharded": true
         },
         "FieldQuery": "select id from `user` where 1 != 1",
-        "Query": "select id from `user` where id = 5 and `name` = 'foo' or id = 12 and `name` = 'bar'",
+        "Query": "select id from `user` where id in ::__vals and (id = 5 or `name` = 'bar') and (`name` = 'foo' or id = 12) and `name` in ('foo', 'bar')",
```

Comment:
Sorry i had a doubt about this re-written query but turns out it looks good to me!